vms/platformvm/state/validator_metadata.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/vms/platformvm/genesis"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+// preDelegateeRewardSize is the size of codec marshalling
+// [preDelegateeRewardMetadata].
+//
+// CodecVersionLen + UpDurationLen + LastUpdatedLen + PotentialRewardLen
+const preDelegateeRewardSize = wrappers.ShortLen + 3*wrappers.LongLen
+
+var _ validatorState = (*metadata)(nil)
+
+type preDelegateeRewardMetadata struct {
+	UpDuration      time.Duration `serialize:"true"`
+	LastUpdated     uint64        `serialize:"true"` // Unix time in seconds
+	PotentialReward uint64        `serialize:"true"`
+}
+
+type validatorMetadata struct {
+	UpDuration               time.Duration `serialize:"true"`
+	LastUpdated              uint64        `serialize:"true"` // Unix time in seconds
+	PotentialReward          uint64        `serialize:"true"`
+	PotentialDelegateeReward uint64        `serialize:"true"`
+
+	txID        ids.ID
+	lastUpdated time.Time
+}
+
+// Permissioned validators originally wrote their values as nil.
+// With Banff we wrote the potential reward.
+// With Cortina we wrote the potential reward with the potential delegatee reward.
+// We now write the uptime, reward, and delegatee reward together.
+func parseValidatorMetadata(bytes []byte, metadata *validatorMetadata) error {
```

Comment:
Would be nice to add some tests for this.

Reply:
[f260cac](https://github.com/ava-labs/avalanchego/pull/1262/commits/f260cace89a196c00079f66af96ba4bc2e1380c3)